### DISCUSSION TITLE
Manage fields types in RESTful API

### DISCUSSION THREAD

**Discussion by Joaquin2000zz** [milvus-io/milvus#33493] [Q&A and General discussion] [answered]
Hello, I'm currently diving into how to use the RESTful API provided by Milvus detailed in https://milvus.io/api-reference/restful/v2.2.x/About.md

In the Create collection section https://milvus.io/api-reference/restful/v2.2.x/Collection%20Operations/create_collection.md
describes how to create a simple collection
`curl --request POST \
     --url "${MILVUS_HOST}:${MILVUS_PORT}/v1/vector/collections/create" \
     --header "Authorization: Bearer ${TOKEN}" \
     --header "accept: application/json" \
     --header "content-type: application/json" \
     -d '{
       "dbName": "default",   
       "collectionName": "medium_articles",
       "dimension": 256,
       "metricType": "L2",
       "primaryField": "id",
       "vectorField": "vector"
      }'`
the problem lies in the lack of schema-type definitions for fields just specifying the vector field.

Also, the description of the parameters supported in the request does not mention how to manage fields that are not the vector field:

<html>
<body>
<h2 style="box-sizing: inherit; margin: 50px 0px 20px; padding: 0px; border: 0px; font-size: 32px; vertical-align: baseline; line-height: 42px; color: rgb(25, 25, 25); font-family: Inter, RobotoMono, SourceCodePro, serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Request</h2><h3 style="box-sizing: inherit; margin: 30px 0px 10px; padding: 0px; border: 0px; font-size: 24px; vertical-align: baseline; line-height: 34px; color: rgb(25, 25, 25); font-family: Inter, RobotoMono, SourceCodePro, serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Parameters</h3><ul style="box-sizing: inherit; margin: 8px 0px; padding: 0px 0px 0px 24px; border: 0px; font-size: 16px; vertical-align: baseline; list-style-type: disc; color: rgb(25, 25, 25); font-family: Inter, RobotoMono, SourceCodePro, serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: inherit; margin: 0px 0px 10px; padding: 0px; border: 0px; font-size: 16px; vertical-align: baseline; line-height: 24px; text-align: left; color: rgb(0, 0, 0); position: relative;"><p style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 16px; vertical-align: baseline; line-height: 24px; font-style: normal; font-weight: 400;">No query parameters required</p></li><li style="box-sizing: inherit; margin: 0px 0px 10px; padding: 0px; border: 0px; font-size: 16px; vertical-align: baseline; line-height: 24px; text-align: left; color: rgb(0, 0, 0); position: relative;"><p style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 16px; vertical-align: baseline; line-height: 24px; font-style: normal; font-weight: 400;">No path parameters required</p></li></ul><h3 style="box-sizing: inherit; margin: 30px 0px 10px; padding: 0px; border: 0px; font-size: 24px; vertical-align: baseline; line-height: 34px; color: rgb(25, 25, 25); font-family: Inter, RobotoMono, SourceCodePro, serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Request Body</h3><pre style="box-sizing: inherit; margin: 24px 0px; padding: 0px 42px 0px 0px; border: 0px; font-size: 13px; vertical-align: baseline; border-radius: 4px; background-color: rgb(246, 246, 246); position: relative; color: rgb(25, 25, 25); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><code class="language-json" style="box-sizing: inherit; margin: 12px 0px; padding: 12px; border: none; font-size: 14px; vertical-align: baseline; line-height: 20px; font-family: SourceCodePro; background-color: transparent; border-radius: 4px; color: rgb(69, 69, 69); display: block; font-weight: 400; width: 844px; overflow-x: auto;"><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">{</span>
    <span class="hljs-attr" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(1, 86, 146);">"collectionName"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">:</span> <span class="hljs-string" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(84, 121, 13);">"string"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">,</span>
    <span class="hljs-attr" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(1, 86, 146);">"dbName"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">:</span> <span class="hljs-string" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(84, 121, 13);">"string"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">,</span>
    <span class="hljs-attr" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(1, 86, 146);">"description"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">:</span> <span class="hljs-string" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(84, 121, 13);">"string"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">,</span>
    <span class="hljs-attr" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(1, 86, 146);">"dimension"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">:</span> <span class="hljs-string" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(84, 121, 13);">"integer"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">,</span>
    <span class="hljs-attr" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(1, 86, 146);">"metricType"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">:</span> <span class="hljs-string" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(84, 121, 13);">"string"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">,</span>
    <span class="hljs-attr" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(1, 86, 146);">"primaryField"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">:</span> <span class="hljs-string" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(84, 121, 13);">"string"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">,</span>
    <span class="hljs-attr" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(1, 86, 146);">"vectorField"</span><span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">:</span> <span class="hljs-string" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline; color: rgb(84, 121, 13);">"string"</span>
<span class="hljs-punctuation" style="box-sizing: inherit; margin: 0px; padding: 0px; border: 0px; font-size: 14px; vertical-align: baseline;">}</span>
<button class="copy-code-btn" style="box-sizing: inherit; background-color: transparent; padding: 0px; margin: 0px; outline: none; border: none; cursor: pointer; display: flex; position: absolute; right: 16px; top: 16px; width: 20px; height: 20px; align-items: center; justify-content: center;"><svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M16 4H6.5C5.11929 4 4 5.11929 4 6.5V16" stroke="#1D2939" strokewidth="2"></path><path d="M19.5 8H8.5C8.22386 8 8 8.22386 8 8.5V19.5C8 19.7761 8.22386 20 8.5 20H19.5C19.7761 20 20 19.7761 20 19.5V8.5C20 8.22386 19.7761 8 19.5 8Z" stroke="#1D2939" strokewidth="2"></path></svg></button></code></pre>

</body>
</html>

Is that supported already? For the moment I'm managing it with a workaround

**Comment by yhmo** [comment, marked as answer]
You are reading the old RESTful API(V1). It only supports creating a collection with 2 fields: "id" and "vector". 

If your Milvus server is the latest v2.3.x or v2.4.x, the V2 RESTful API is recommended: https://milvus.io/api-reference/restful/v2.4.x/v2/Collection%20(v2)/Create.md

**Reply by Joaquin2000zz** [reply]
Thank you so much! Have a great day!
